src/app/sportovni-akce/[category]/[id]/page.tsx

```diff
@@ -0,0 +1,178 @@
+import { NextPage } from 'next';
+import Image from 'next/image';
+import Link from 'next/link';
+import { notFound } from 'next/navigation';
+
+import { EventDetail } from './EventDetail';
+import { OwnerCard } from './OwnerCard';
+import { Events } from '../../../../components/Events';
+import { getAllEvents } from '../../../../api/events';
+import { getEvent } from '../../../../api/events';
+import { StarRating } from '../../../../components/StarRating';
+import { mockEventOwner } from './mock';
+import { Routes, defaultOwner, levelLabels } from '../../../../utils/constants';
+import { getImagePath, getSportLabel } from '../../../../utils/functions';
+import { formatDate, formatTime } from '../../../../utils/dateUtils';
+import { EmailForm } from '../../../../components/EmailForm';
+import { slugifyCategory } from '../../../../utils/slugifyUtils';
+
+type EventPageProps = {
+  params: {
+    category: string;
+    id: string;
+  };
+};
+
+const EventPage: NextPage<EventPageProps> = async ({ params }) => {
+  const events = (await getAllEvents()) || [];
+
+  const { category, id } = params;
+
+  const event = await getEvent(id);
+
+  if (!event) {
+    return notFound();
+  }
+
+  const { name, date, location, sport, description, level, owner } = event;
+  const eventOwner = owner || defaultOwner;
+
+  const sportLabel = getSportLabel(sport);
+  const levelLabel = levelLabels[level];
+
+  const expectedCategory = slugifyCategory(
+    `${sportLabel} ${location} ${levelLabel}`
+  );
+
+  if (category !== expectedCategory) {
+    return notFound();
+  }
+
+  const formattedDate = formatDate(date);
+  const formattedTime = formatTime(date);
+
+  return (
+    <>
+      <div className="relative mt-14 flex flex-col items-start px-4 md:flex-row md:px-0 lg:mt-0">
+        <div className="relative flex h-full w-full flex-col items-center justify-center pr-0 lg:pr-14">
+          <section className="relative w-full">
+            <figure className="relative aspect-3/1 h-full w-full overflow-hidden">
+              <Image
+                src={getImagePath(id, sport)}
+                alt="event image"
+                className="rounded-md object-cover"
+                sizes="auto"
+                fill
+                priority
+              />
+            </figure>
+            <h1 className="mt-12 text-center text-2xl font-medium leading-normal md:mt-9 lg:text-start lg:text-4xl">
+              {name}
+            </h1>
+            <p className="mt-8 text-lg font-light">{description}</p>
+            <EventDetail
+              className="mx-auto mt-8 lg:hidden"
+              event={event}
+              formattedDateTime={{ formattedDate, formattedTime }}
```

Comment:
Ah!  React will do a comparison on the props to determine if the component needs to re-render... and issue with objects: every time that component renders, { formattedDate, formattedTime } creates a new object with new references... even if formattedDate and formattedTime haven't changed - so  React.memo would cause some unnecessary re-renders.